### PR TITLE
NIFI-3133: Ensure correct checks against Remote Group Ports

### PR DESCRIPTION
**Pull request by mcgilman**
- Ensuring that Remote Group Ports are always authorized according to their Remote Process Group.

### CODE REVIEW

**Comment by JPercivall** [comment]
Reviewing

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/FlowController.java [hunk 1]
```diff
@@ -4034,19 +4034,29 @@ public Authorizable createDataAuthorizable(final String componentId) {
...
         return authorizable;
     }
 
     @Override
+    public Authorizable createRemoteDataAuthorizable(String remoteGroupPortId) {
+        final DataAuthorizable authorizable;
+
+        final RemoteGroupPort remoteGroupPort = getRootGroup().findRemoteGroupPort(remoteGroupPortId);
+        if (remoteGroupPort == null) {
+            throw new ResourceNotFoundException("The component that generated this event is no longer part of the data flow.");
```
Comment (JPercivall) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/FlowController.java]: I think there is a bit of copy paste error with this Exception message. My understanding is that this method is not limited to an "event".

Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/FlowController.java]: With the refactoring it is just events. However, there is still one spot that is unused that can now be removed. I'll update.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/controller/ControllerFacade.java [hunk 2]
```diff
@@ -1432,13 +1452,20 @@ public int compare(AttributeDTO a1, AttributeDTO a2) {
...
-        final Connectable connectable = root.findConnectable(dto.getComponentId());
+        final Connectable connectable = root.findLocalConnectable(dto.getComponentId());
         if (connectable != null) {
             dto.setGroupId(connectable.getProcessGroup().getIdentifier());
             dto.setComponentName(connectable.getName());
             return;
         }
 
+        final RemoteGroupPort remoteGroupPort = root.findRemoteGroupPort(dto.getComponentId());
+        if (connectable != null) {
```
Comment (JPercivall) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/controller/ControllerFacade.java]: Looks like a copy/paste error, this should check "remoteGroupPort"

**Comment by JPercivall** [comment]
+1

Visually verified code and did a contrib-check build. In a secure 3 node cluster ran through a number of tests involving RPGs, Input/Output ports and policies. I also ran a standalone instance to verify nothing there regressed. Everything worked as expected. 

Thanks @mcgilman I will squash and merge.

**Pull request closed without merging**